server.js

```diff
@@ -6537,6 +6538,61 @@ cache((data, match, sendBadge, request) => {
   });
 }));
 
+// User defined sources - JSON or XML response
+camp.route(/^\/(json|xml)\/([^/]+)\/([^/]+)\/(https?\:\/\/.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(query, match, sendBadge, request) {
+  var type = match[1];
+  var vars = match[2].replace(/--/g,'%2D').split('-');
+  var label = vars[0];
+  var color = vars[1] || 'brightgreen';
+  var prefix = vars[2] || '';
+  var suffix = vars[3] || '';
+  var pathExpression = decodeURI(match[3]);
+  var userURI = match[4];
+  var format = match[5];
+
+  // API URL
+  var apiUrl = encodeURI(userURI);
+
+  var badgeData = getBadgeData(decodeURI(label), query);
+  request(apiUrl, {json:true}, function(err, res, data) {
+    if (res && res.statusCode === 404) {
+      badgeData.text[1] = 'invalid server';
+      sendBadge(format, badgeData);
+      return;
+    }
+    if (err != null || !res || res.statusCode !== 200) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+
+      if (type === 'xml') {
+        xml2js.parseString(data.toString(), function (err, json) {
+          if (err) throw (err);
+          data = json;
+        });
+      } else {
+        data = (typeof data == 'object' ? data : JSON.parse(data));
+      }
+
+      badgeData.text[1] = (prefix || "") + jp.query(data, pathExpression).join(", ") + (suffix || "");
+
+      if (sixHex(color)) {
+        badgeData.colorB = '#' + color;
+      } else {
+        badgeData.colorscheme = color;
+      }
```

Comment:
Rather than handle the color in the URL, could we just use `?colorB=`? That way the interface is a bit simpler.

I'd really like that to support named colors, though we should solve that in a global way. If you're game to look into that that would be 👍, though no pressure of course! (Should be a separate PR.)

Reply:
No worries, will make a PR at some point 😄